### PR TITLE
Changes and additions to enable text truncation of the project menu and username at primary media query breakpoints

### PR DESCRIPTION
**Pull request by sg00dwin**
@spadgett 
cc @benjaminapetersen

### CODE REVIEW

**Comment by sg00dwin** [comment]
At 1600 the project-menu will display 60 characters before truncating.

**1200**
desktop1200

**ipad landscape**
ipad-landscape

**ipad portrait**
ipad-portrait

**iphone6**
iphone6

**iphone5**
iphone5

File: assets/app/styles/_project-menu.less [hunk 1]
```diff
@@ -138,10 +147,10 @@
     float: left;
     margin: 0;
     padding-top: 0;
-    max-width: 480px;
+    max-width: 450px; // Default project menu width
```
Comment (sg00dwin) [inline assets/app/styles/_project-menu.less]: Adjusted down to cover IE

File: assets/app/styles/_navbar-alt.less [hunk 2]
```diff
@@ -158,6 +158,7 @@
     .align-items(@align: center);
     .flex(@columns: 1 0 0%);
     .justify-content(@justify: center);
+    width: 100%;
```
Comment (sg00dwin) [inline assets/app/styles/_navbar-alt.less]: Fixes IE10 issue

File: assets/app/styles/_project-menu.less [hunk 3]
```diff
@@ -31,7 +31,7 @@
       box-shadow: none;
       border-top: 0;
       line-height: @line-height-base;
-      padding: 7px 45px 7px 20px;
+      padding: 7px 50px 7px 20px;
```
Comment (sg00dwin) [inline assets/app/styles/_project-menu.less]: This is so the ellipsis doesn't extend into down icon

File: assets/app/styles/_project-menu.less [hunk 4]
```diff
@@ -104,6 +101,12 @@
   }
 }
 
+.navbar-project-menu .bootstrap-select.btn-group .btn .filter-option {
+  max-width: 200px; // Width at 320px
+  min-width: 100%; // Now make menu full width and truncate at mobile
```
Comment (spadgett) [inline assets/app/styles/_project-menu.less]: I'm having trouble following this. If you have min-width: 100%, do you need max-width set at all?

File: assets/app/styles/_project-menu.less (hunk 3 above)
Comment (spadgett) [inline assets/app/styles/_project-menu.less]: Better to add comments like this to the code than as comments to the PR.

Comment (benjaminapetersen) [inline assets/app/styles/_project-menu.less]: Agree!

File: assets/app/styles/_project-menu.less (hunk 4 above)
Comment (benjaminapetersen) [inline assets/app/styles/_project-menu.less]: Curious about that as well.   Not sure if we need 5 selectors to hit the `.filter-option`?

File: assets/app/styles/_navbar-alt.less (hunk 2 above)
Comment (benjaminapetersen) [inline assets/app/styles/_navbar-alt.less]: This could be helpful to comment in code as well:
```css
width: 100%; // fix ie10 issue
```

**Comment by sg00dwin** [comment]
@spadgett rebased

Chrome needs a max-width in order to truncate at mobile and 200px is the available space for the project name at 320px, if it's longer it will be truncated in order to display the add to project '+' icon.

[screen shot 2016-02-15 at 4 19 10 pm
](https://drive.google.com/a/redhat.com/file/d/0B3F2EMy0q8RjTmlHN0ZsSnBfZ1U/view?usp=sharing)

200px-max-width

**Comment by benjaminapetersen** [comment]
@sg00dwin works for me!

**Comment by spadgett** [comment]
[merge]

**Comment by spadgett** [comment]
@sg00dwin Changing the `h1` broke the e2e tests. Let's catch up on that.

**Comment by benjaminapetersen** [comment]
@spadgett @sg00dwin ah tests... I wanna fix that.  Would be great to have some kind of a `page-title` class or whatever, and you can put it on any tag & tests (both ours & QE) can hook into that.  

Per the [discussion on this issue](https://github.com/openshift/origin/issues/6746), good hooks & markup changes not breaking tests.

**Comment by sg00dwin** [comment]
[test]

**Comment by sg00dwin** [comment]
@spadgett is this failure related to the h1 issue?

**Comment by spadgett** [comment]
Looks unrelated.

[test]

**Comment by spadgett** [comment]
[merge]

**Pull request merged**